zerver/tests/test_invite.py

```diff
@@ -997,6 +999,50 @@ def test_multi_user_invite(self) -> None:
             ]
         )
 
+    def test_direct_notification_for_accepted_invitation(self) -> None:
+        """
+        New test to check if notification-bot message is sent to the referrer
+        when notify_referrer_on_join is false.
+        """
+
+        self.login("hamlet")
+        user_profile = self.example_user("hamlet")
+        invitee = self.nonreg_email("alice")
+        realm = get_realm("zulip")
+        self.assert_json_success(self.invite(invitee, ["Denmark"], False))
+        self.assertTrue(find_key_by_email(invitee))
+        self.submit_reg_form_for_user(invitee, "password")
+
+        assert user_profile.recipient_id is not None
+        invite_acceptance_notification_message = Message.objects.filter(
+            recipient_id=user_profile.recipient_id, realm=realm
+        ).last()
+
+        self.assertIsNone(
+            invite_acceptance_notification_message, "Unexpected referrerr message found."
```

Comment:
I don't understand this check; `"Unexpected referrerr message found."` has a pretty weird typo, and also, it seems like it might be possible for the user to have received some unrelated DM previously?